src/components/form/field_number/field_number.tsx

```diff
@@ -39,7 +39,7 @@ export type EuiFieldNumberProps = InputHTMLAttributes<HTMLInputElement> &
     readOnly?: boolean;
     min?: number;
     max?: number;
-    step?: number;
+    step?: number | 'any';
```

Comment:
Because our docsgen has problems with TS types, the props table renders this prop as:
<img width="333" alt="Screen Shot 2020-06-05 at 11 12 46 AM" src="https://user-images.githubusercontent.com/549577/83893147-95f56900-a71d-11ea-8460-be41f0f27a82.png">

We need to add a comment so they know what "enum" is allowed
```suggestion
    /**
     * Specifies the granularity that the value must adhere to.
     * Accepts a `number` or the string `'any'` for no stepping to allow for any value.
     * Defaults to `1`
     */
    step?: number | 'any';
```